### DISCUSSION TITLE
Strict resolvers types + expensive field resolvers

### DISCUSSION THREAD

**Discussion by GraemeF** [Urigo/graphql-modules#2226] [Q&A]
Hello! I'm trying to enable [strict resolvers types](https://www.graphql-modules.com/docs/essentials/type-safety#strict-resolvers-types) as described in [this blog post](https://www.the-guild.dev/blog/better-type-safety-for-resolvers-with-graphql-codegen), but I'm required to return an object with all of the non-nullable fields resolved. Many of my field resolvers are expensive (they go off to another API or data source) and so I want to lazily resolve those fields only if requested, leaning on Dataloader to reduce the amount of work needed.

In other words:

```graphql
type Query {
  foos: [Foo!]!
}

type Foo {
  id: ID!
  someExpensiveField: String!
}
```

From my `Query.foos` resolver I'd like to be able to return some `Foo` instances that only have an `id` and not `someExpensiveField`, and have the `Foo.someExpensiveField` do the work only if necessary, but the generated types won't allow a partial `Foo` to be returned.

Is there a nice way to do this without making all fields optional (which wouldn't make sense for my schema)?
